### PR TITLE
[ML] data frame analytics inference improvements

### PR DESCRIPTION
**Pull request by benwtrent**
this commit prevents the model from being cached unnecessarily.
it also keeps the model bytes from being removed from the circuit breaker
until inference is complete.

### CODE REVIEW

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/inference/InferenceRunner.java [hunk 1]
```diff
@@ -73,10 +73,12 @@ public void run(String modelId) {
         LOGGER.info("[{}] Started inference on test data against model [{}]", config.getId(), modelId);
         try {
             PlainActionFuture<LocalModel> localModelPlainActionFuture = new PlainActionFuture<>();
-            modelLoadingService.getModelForSearch(modelId, localModelPlainActionFuture);
+            modelLoadingService.getModelForPipeline(modelId, localModelPlainActionFuture);
             LocalModel localModel = localModelPlainActionFuture.actionGet();
             TestDocsIterator testDocsIterator = new TestDocsIterator(new OriginSettingClient(client, ClientHelper.ML_ORIGIN), config);
-            inferTestDocs(localModel, testDocsIterator);
+            try (localModel) {
```
Comment (przemekwitek) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/inference/InferenceRunner.java:79]: Would it make sense to change this line to:
```
try (LocalModel localModel = modelLoadingService.getModelForPipeline(modelId, localModelPlainActionFuture).actionGet()) {
```
?

Comment (benwtrent) [reply x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/inference/InferenceRunner.java]: I did a slight refactor to take this comment into account. 

Ultimately, it does not really matter.

**Review by przemekwitek — APPROVED** [review]
LGTM

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Pull request merged**